cmake/FindR.cmake

```diff
@@ -46,42 +46,69 @@ if(R_COMMAND)
   set(OLD_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
   set(CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES} ".dll")
 
-  execute_process(WORKING_DIRECTORY .
-                  COMMAND ${R_COMMAND} RHOME
-                  OUTPUT_VARIABLE R_ROOT_DIR
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-  set(R_HOME ${R_ROOT_DIR} CACHE PATH "R home directory obtained from R RHOME")
-  
-  execute_process(WORKING_DIRECTORY .
-                  COMMAND ${R_COMMAND} CMD config --ldflags
-                  OUTPUT_VARIABLE R_LDFLAGS
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-  set(R_LDFLAGS ${R_LDFLAGS} CACHE PATH "R CMD config --ldflags")
-
-  find_path(R_INCLUDE_DIR R.h
-            HINTS ${R_ROOT_DIR} ${R_ROOT_DIR}/bin/${R_LIB_ARCH}
-            PATHS /usr/local/lib /usr/local/lib64 /usr/share
-            PATH_SUFFIXES include R/include
-            DOC "Path to file R.h")
-
-  find_library(R_LIBRARY_BASE R
-            HINTS ${R_ROOT_DIR}/lib ${R_ROOT_DIR}/bin/${R_LIB_ARCH}
-            DOC "R library (example libR.a, libR.dylib, etc.).")
-
-  find_library(R_LIBRARY_BLAS NAMES Rblas blas
-            HINTS ${R_ROOT_DIR}/lib ${R_ROOT_DIR}/bin/${R_LIB_ARCH}
-            DOC "Rblas library (example libRblas.a, libRblas.dylib, etc.).")
-
-  find_library(R_LIBRARY_LAPACK NAMES Rlapack lapack
-            HINTS ${R_ROOT_DIR}/lib ${R_ROOT_DIR}/bin/${R_LIB_ARCH}
-            DOC "Rlapack library (example libRlapack.a, libRlapack.dylib, etc.).")
-
-  find_library(R_LIBRARY_READLINE readline
-            DOC "(Optional) system readline library. Only required if the R libraries were built with readline support.")
+  if(CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
+    message(STATUS "Configuring for Emscripten...")
+
+    set(R_HOME "${CMAKE_PREFIX_PATH}/lib/R" CACHE PATH "R home directory for Emscripten")
+
+    set(R_INCLUDE_DIR "${R_HOME}/include" CACHE PATH "Path to R include directory")
+
+    execute_process(WORKING_DIRECTORY .
+                    COMMAND ${R_COMMAND} CMD config --ldflags
+                    OUTPUT_VARIABLE R_LDFLAGS
+                    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    set(R_LDFLAGS ${R_LDFLAGS} CACHE PATH "R CMD config --ldflags")
+    message(STATUS "R_LDFLAGS obtained from R command: ${R_LDFLAGS}")
+
+    set(R_LIBRARY_BASE "${R_HOME}/lib/libR.so" CACHE FILEPATH "R library (libR.so) in Emscripten")
+
+    set(R_LIBRARY_BLAS "${R_HOME}/lib/libRblas.so" CACHE FILEPATH "Rblas library (libRblas.so) in Emscripten")
+
+    set(R_LIBRARY_LAPACK "${R_HOME}/lib/libRlapack.so" CACHE FILEPATH "Rlapack library (libRlapack.so) in Emscripten")
```

Comment:
Possibly not the best way to address this but for some reason `R_COMMAND` doesn't give back the correct `R_HOME` and subsequently `R_INCLUDE_DIR` , `R_LIBRARY_BASE` (which are found in relation to `R_HOME`) aren't being located.

Hence I split this as emscripten vs non-emscripten. 
Now I think we have the correct locations

Reply:
```
-- Configuring for Emscripten...
-- R_HOME           = /home/runner/micromamba/envs/xeus-r-wasm-host/lib/R
-- R_INCLUDE_DIR    = /home/runner/micromamba/envs/xeus-r-wasm-host/lib/R/include
-- R_LDFLAGS        = -Wl,--export-dynamic -fopenmp -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -L/lib/R/lib -lR -lpcre2-8 -ldeflate -llzma -lbz2 -lz -ltirpc -lrt -ldl -lm -licuuc -licui18n
-- R_LIBRARY_BASE   = /home/runner/micromamba/envs/xeus-r-wasm-host/lib/R/lib/libR.so
-- R_LIBRARY_BLAS   = /home/runner/micromamba/envs/xeus-r-wasm-host/lib/R/lib/libRblas.so
-- R_LIBRARY_LAPACK = /home/runner/micromamba/envs/xeus-r-wasm-host/lib/R/lib/libRlapack.so
```

Reply:
Was just thinking why the `R_HOME` is set as `/lib/R` where as we want is to be something like "env/lib/R"  and I guess the [pre.js](https://github.com/emscripten-forge/recipes/blob/main/recipes/recipes_emscripten/r-base/pre.js) sets it so.

Reply:
> Was just thinking why the `R_HOME` is set as `/lib/R` where as we want it to be something like "env/lib/R". Does that spawn from the [pre.js](https://github.com/emscripten-forge/recipes/blob/main/recipes/recipes_emscripten/r-base/pre.js) ?

[This patch](https://github.com/emscripten-forge/recipes/blob/ff555adf67f4cc6356db84a59076ac2042074013/recipes/recipes_emscripten/r-base/patches/0008-Set-default-R_Home.patch#L20) also sets the default value of `R_HOME` to be *lib/R*. However, `R_HOME` should be overwritable depending on the use case.

Reply:
> Possibly not the best way to address this but for some reason `R_COMMAND` doesn't give back the correct `R_HOME` when building against emscripten and subsequently `R_INCLUDE_DIR` , `R_LIBRARY_BASE` (which are found in relation to `R_HOME`) aren't being located.
> 
> Hence I split this as emscripten vs non-emscripten. Now I think we have the correct locations

From the CI:
```
-- R_COMMAND found at: /home/runner/micromamba/envs/xeus-r-wasm-host/bin/R
-- R_SCRIPT_COMMAND found at: /home/runner/micromamba/envs/xeus-r-wasm-host/bin/Rscript
```
bin/R and bin/Rscript are shell wrappers for the corresponding binaries. We shouldn't use these during building because they'll try to execute R.js and Rscript.js. If we need an R executable, we should add `r-base` to the build environment. This could be one of the reasons it does not give the correct value of `R_HOME`, but I'm not sure what values it was giving before.

Reply:
Argg I think I squashed the commits but R_HOME was found to be "/lib/R" (didn't show any prefix related to it) and other variables weren't found.

Will go through the above comments soon !